### DISCUSSION TITLE
Hosting Open WebUI in Azure App Service only for Azure Open AI GPT Models (No Local Ollama)

### DISCUSSION THREAD

**Discussion by KingstonJoseph12** [open-webui/open-webui#4995] [General]
Hello everyone,

My organization has requested a chatbot solution that allows users to either upload documents and interact with them in a conversational manner (using RAG) or ask general questions based on pre-uploaded documents. I am currently working on developing this solution using Azure OpenAI with GPT-4 and GPT-4o models and deploying it directly through Azure Web App. For reference, I'm using this Azure Default app: [Sample-APP-AOAI-chatGPT](https://github.com/microsoft/sample-app-aoai-chatGPT).

I have used Open WebUI to run Ollama models on my personal PC, and I've found that Open WebUI offers a superior user interface and better capabilities compared to other tools I've tried.

I'm interested in deploying Open WebUI in a way that integrates with Azure App Service and utilizes Microsoft AD Group Authentication. Does anyone have experience with this setup, or know if it's possible to connect Open WebUI with Azure App Service and enable Microsoft AD Group Authentication seamlessly?

Any guidance or suggestions would be greatly appreciated! We are also open to have a monthly sponsorship with Open WebUI if this solution is deployed.

**Comment by Pav-Bahia** [comment]
I this maybe what your looking for, using the azure hosted openai

https://blakedrumm.com/blog/azure-container-apps-openweb-ui/

**Reply by KingstonJoseph12** [reply]
I successfully did it. Thank you.

**Reply by lakergreat1** [reply]
Would you mind providing instructions?  I'm looking to do the same thing!

Thank you!

**Reply by KingstonJoseph12** [reply]
The process is almost identical to what’s described in the link above, which provides detailed instructions. After building your image, push it to the container registry. Once done, you can deploy the container to Azure App Services, where you’ll also need to configure the necessary environment variables.

**Reply by HMFlol** [reply]
> The process is almost identical to what’s described in the link above, which provides detailed instructions. After building your image, push it to the container registry. Once done, you can deploy the container to Azure App Services, where you’ll also need to configure the necessary environment variables.

Hey there. Sorry to respond to this over a month later, but I'm wondering what methods you used to integrate azure open ai models as well as utilize authentication using Entra? Looking to do the same. Currently using LiteLLM as a proxy for the Azure OpenAI models, but there is likely a better way.

**Reply by KingstonJoseph12** [reply]
Hi Ryne,

Here’s the step-by-step process I followed:

1. **Pulled the Base Image**  
   I started by pulling the image:  
   ```bash
   docker pull ghcr.io/open-webui/open-webui:main
   ```

2. **Customized the Image**  
   I tagged the image to create a customized version, where I removed some visual elements to simplify it (I deployed it for my dad, who is a school teacher, so simplicity was key):  
   ```bash
   docker tag ghcr.io/open-webui/open-webui:main myrepo/mywebui:latest
   ```

3. **Logged in to Azure CLI**  
   On my Windows PC, I logged in to the Azure CLI:  
   ```bash
   az acr login --name containerregistryname.azurecr.io
   ```

4. **Pushed the Customized Container to Azure Container Registry**  
   After creating an Azure Container Registry from the Azure web portal, I pushed my customized container:  
   ```bash
   docker push containerregistryname.azurecr.io/open-webui:main
   ```

5. **Created an Azure Web App**  
   Finally, I created an Azure Web App by connecting it to the container registry and configured the required environment variables through the Azure Web App UI.  
   *(For enabling Microsoft Entra, you can follow [these instructions](https://docs.openwebui.com/features/sso#microsoft)).*

Let me know if you need any further details! You can [_reach out to me_](https://www.linkedin.com/in/kingston-joseph/).

**Reply by HMFlol** [reply]
Thanks for the reply!

Did you deploy it with pipelines as well for integration with your azure openai models? Or did you integrate those a different way?

Cheers!

**Reply by KingstonJoseph12** [reply]
No, I did not use pipelines, I used Functions. You can find the code here.

https://openwebui.com/f/nomppy/azure/

**Reply by auliaihendradi** [reply]
If you have LiteLLM it can also consolidate all your Azure OpenAI endpoints, currently i used it for gpt4o, text-embedding-3-large, and dall-e, so in the system you just put litellm endpoint instead of openai endpoint

**Comment by McNickSistoPro** [comment]
How did you connect it with Ollama ? @KingstonJoseph12

**Reply by McNickSistoPro** [reply]
@KingstonJoseph12 also which file do you drop those variables in ? Did you do it through the interface ?
![image](https://github.com/user-attachments/assets/9058b028-95af-453d-8980-7ba38b0efc79)
